bones/numericBone.py

```diff
@@ -55,12 +55,13 @@ def singleValueFromClient(self, value, skel, name, origData):
 				value = int(rawValue)
 			else:
 				return self.getDefaultValue(), [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, name, "Invalid Value")]
+		if not value:
```

Comment:
This will also disallow `0` (zero) as value.

Reply:
That's the expected behavior. Zero is the numerical equivalent of the empty string. If the field is required, you'll have to provide a non-zero value.

Reply:
What? 
With this kind of validation we're will not able to store in a tournament software the scores of a game which ends 2:0 (except we remove the required flag, which is stupid for our data completeness). And we are not able to start a counter from zero (except we use not the validation/fromClient). In a online-shop-form you're not able to select "0" as amount for e.g. extra screws. In a accounting software you're not able to set the annual profit to 0 €.
Sorry, but this is not realistic and I see there no reason why we should handle zero as empty string, it's a valid and normal digit in this context! In a numeric-input you can submit a empty string as well - which will be of course invalid for a required-bone.

Reply:
There where other objections as well. We'll defer this to the next ViUR Meeting. I think there is a issue with the meaning of required. It does not mean "enforce there is a value in the skeleton/datastore" - it now always is. For numeric-bones zero is the default (unless overwritten) and it will serve the add form prefilled with it. Maybe required is better understood as "filled" or "non-empty". Maybe we add an option of what value is treated as empty - so zero is valid and -999 isn't?

Reply:
This should be done with the `isEmpty` and `getEmptyValue` methods.